src/pages/index.js

```diff
@@ -0,0 +1,264 @@
+  evt.preventDefault();
+  const submitbtn = evt.submitter;
+  setButtonText(submitbtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      closeModal(deleteCardModal);
+      selectedCard.remove();
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitbtn, false, "Delete", "Deleting...");
+    });
+}
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-button_active");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button_active");
+    })
+    .catch(console.error);
+}
+//get card function
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  cardNameEl.textContent = data.name;
+
+  const cardLinkEl = cardElement.querySelector(".card__image");
+  cardLinkEl.src = data.link;
+  cardLinkEl.alt = data.name;
+  const likeSatus = data.isLiked;
+
+  const cardlikeBtn = cardElement.querySelector("#likeButton");
+  if (likeSatus === true) {
+    cardlikeBtn.classList.add("card__like-button_active");
+  }
+
+  cardlikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  const deleteBtn = cardElement.querySelector("#deleteBtn");
+
+  deleteBtn.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  cardLinkEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageElement.src = data.link;
+    previewModalImageElement.alt = data.name;
+    previewModalCaptionElement.textContent = data.name;
+  });
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+  modal.addEventListener("click", handleOverlayClick);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+  modal.removeEventListener("click", handleOverlayClick);
+}
+function handleOverlayClick(event) {
+  if (event.target.classList.contains("modal_opened")) {
+    closeModal(event.target);
+  }
+}
+
+function handleEscapeKey(event) {
+  if (event.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    if (modal) {
+      closeModal(modal);
+    }
+  }
+}
+
+//listeners
+profileEditButton.addEventListener("click", () => {
+  openModal(editModal);
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(editModal, config);
+});
+
+cardModelButton.addEventListener("click", () => {
+  openModal(cardModal);
+  resetValidation(cardModal, config);
+});
+
+avatarModelButton.addEventListener("click", () => {
+  openModal(avatarModal);
+  resetValidation(avatarForm, config);
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs.  


It's a waste of resources if you call it for the card and avatar forms. And you don't need to remove errors from the inputs  of the card and avatar forms.